src/swap/handle_get_printable_amount.c

```
@@ -0,0 +1,54 @@
+#ifdef HAVE_SWAP
+
+#include <string.h>  // memset, explicit_bzero
+#include "handle_swap_sign_transaction.h"
+#include "swap.h"
+#include "os.h"
+#include "format_bigint.h"
+#include "constants.h"
+
+/* Set empty printable_amount on error, printable amount otherwise */
+void swap_handle_get_printable_amount(get_printable_amount_parameters_t* params) {
+    PRINTF("coin_configuration %.*H\n",
+           params->coin_configuration_length,
+           params->coin_configuration);
+    PRINTF("amount %.*H\n", params->amount_length, params->amount);
+    uint8_t decimals;
+    char ticker[MAX_SWAP_TOKEN_LENGTH] = {0};
+
+    PRINTF("Inside Ton swap_handle_get_printable_amount\n");
+
+    // If the amount is a fee, its value is nominated in TON even if we're doing an TRC20 swap
+    // If there is no coin_configuration, consider that we are doing a TON swap
+    if (params->is_fee || params->coin_configuration == NULL) {
```

Aren't fees supposed to be fixed -not encoded inside the tx- in TON? In this case, we should raise an error if we get `is_fee == true`.